### PR TITLE
gh-117618: Make package.module searchable for breakpoints and clean up docs

### PR DESCRIPTION
**Pull request by gaogaotiantian**
* Make `package.module` searchable by normalizing `filename` first in `lookupmodule`.
* Clean up the code in `lookupmodule`.
* Clarify the docs for `break`. Re-structured to make it more reasonable.

* Issue: gh-117618

----
📚 Documentation preview 📚: https://cpython-previews--117619.org.readthedocs.build/

### CODE REVIEW

**Comment by gaogaotiantian** [comment]
Hi @iritkatriel , let me know if you need more explanation for this PR. The motivation is in the linked issue.

**Comment by gaogaotiantian** [comment]
Hey @iritkatriel , a friendly ping here, as it's getting closer to the feature freeze. I think this makes the `break` command more consistent and it would be nice to have it in 3.13 instead of 3.14.

File: Misc/NEWS.d/next/Library/2024-04-08-03-23-22.gh-issue-117618.-4DCUw.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Support ``package.module`` as ``filename`` for ``break`` command
```
Comment (iritkatriel) [inline Misc/NEWS.d/next/Library/2024-04-08-03-23-22.gh-issue-117618.-4DCUw.rst]: Mention pdb

Comment (gaogaotiantian) [reply Misc/NEWS.d/next/Library/2024-04-08-03-23-22.gh-issue-117618.-4DCUw.rst]: Is the current version better?

File: Doc/library/pdb.rst [hunk 2]
```diff
@@ -328,12 +328,16 @@ can be overridden by the local file.
 
 .. pdbcommand:: b(reak) [([filename:]lineno | function) [, condition]]
 
-   With a *lineno* argument, set a break there in the current file.  With a
-   *function* argument, set a break at the first executable statement within
-   that function.  The line number may be prefixed with a filename and a colon,
+   With a *lineno* argument, set a break at line *lineno* in the current file.
+   The line number may be prefixed with a *filename* and a colon,
    to specify a breakpoint in another file (probably one that hasn't been loaded
```
Comment (iritkatriel) [inline Doc/library/pdb.rst]:
```suggestion
   to specify a breakpoint in another file (possibly one that hasn't been loaded
```

File: Doc/library/pdb.rst [hunk 3]
```diff
@@ -328,12 +328,16 @@ can be overridden by the local file.
...
-   yet).  The file is searched on :data:`sys.path`.  Note that each breakpoint
-   is assigned a number to which all the other breakpoint commands refer.
+   yet).  The file is searched on :data:`sys.path`.  Accepatable form of *filename*
```
Comment (iritkatriel) [inline Doc/library/pdb.rst]:
```suggestion
   yet).  The file is searched on :data:`sys.path`.  Accepatable forms of *filename*
```

File: Doc/library/pdb.rst [hunk 4]
```diff
@@ -328,12 +328,16 @@ can be overridden by the local file.
...
+   includes ``/abspath/to/file.py``, ``relpath/file.py``, ``module`` or
```
Comment (iritkatriel) [inline Doc/library/pdb.rst]:
```suggestion
   are ``/abspath/to/file.py``, ``relpath/file.py``, ``module`` and
```

File: Doc/library/pdb.rst [hunk 5]
```diff
@@ -342,6 +346,9 @@ can be overridden by the local file.
    of times that breakpoint has been hit, the current ignore count, and the
    associated condition if any.
 
+   Note that each breakpoint is assigned a number to which all the other
```
Comment (iritkatriel) [inline Doc/library/pdb.rst]:
```suggestion
   Each breakpoint is assigned a number to which all the other
```

**Comment by gaogaotiantian** [comment]
Thanks for the doc fix!

**Review by iritkatriel — APPROVED** [review]

**Pull request merged**